core/state_transition.go

```diff
@@ -452,7 +464,7 @@ func (st *stateTransition) execute() (*ExecutionResult, error) {
 
 		st.state.RevertToSnapshot(snap)
 		// Even though we revert the state changes, always increment the nonce for the next deposit transaction
-		st.state.SetNonce(st.msg.From, st.state.GetNonce(st.msg.From)+1)
+		st.state.SetNonce(st.msg.From, st.state.GetNonce(st.msg.From)+1, tracing.NonceChangeEoACall)
```

Comment:
Needed to add a nonce-change reason for tracing - is EoACall the best reason to pick here?

Reply:
If it's a contract-initiated tx, then the sender address gets masked, and so practically an EOA in L2. Since it's only tracing metadata I think that is sufficient. We can change it later if we see a need to.